tests/test_utils/gcp_system_helpers.py

```diff
@@ -68,3 +77,92 @@ def provide_gcp_context(
     return provide_gcp_conn_and_credentials(
         key_file_path=key_file_path, scopes=scopes, project_id=project_id
     )
+
+
+class GoogleSystemTest(SystemTest):
+    @staticmethod
+    def _project_id():
+        return os.environ.get("GCP_PROJECT_ID")
+
+    @staticmethod
+    def _service_key():
+        return os.environ.get(CREDENTIALS)
+
+    @contextmanager
+    def authentication(self):
+        self.authenticate()
+        yield
+        self.revoke_authentication()
+
+    @classmethod
+    def authenticate(cls):
+        """
+        Authenticate with service account specified via key name.
+        Required only when we use gcloud / gsutil.
+        """
+        cls.executor.execute_cmd(
+            [
+                "gcloud",
+                "auth",
+                "activate-service-account",
+                f"--key-file={cls._service_key()}",
+                f"--project={cls._project_id()}",
+            ]
+        )
+
+    @classmethod
+    def revoke_authentication(cls):
+        """
+        Change default authentication to none - which is not existing one.
+        """
+        cls.executor.execute_cmd(
+            [
+                "gcloud",
+                "config",
+                "set",
+                "account",
+                "none",
+                f"--project={cls._project_id()}",
+            ]
+        )
+
+    @classmethod
+    def execute_with_ctx(cls, cmd: List[str], key: str = GCP_GCS_KEY):
+        """
+        Executes command with context created by provide_gcp_context and activated
+        service key.
+        """
+        with provide_gcp_context(key):
+            cls.authenticate()
+            env = os.environ.copy()
+            cls.executor.execute_cmd(cmd=cmd, env=env)
+            cls.revoke_authentication()
+
+    @classmethod
+    def create_gcs_bucket(cls, name: str, location: Optional[str] = None) -> None:
+        cmd = ["gsutil", "mb"]
+        if location:
+            cmd += ["-c", "regional", "-l", location]
+        cmd += [f"gs://{name}"]
+        cls.execute_with_ctx(cmd, key=GCP_GCS_KEY)
+
+    @classmethod
+    def delete_gcs_bucket(cls, name: str):
+        cmd = ["gsutil", "-m", "rm", "-r", f"gs://{name}"]
+        cls.execute_with_ctx(cmd, key=GCP_GCS_KEY)
+
+    @classmethod
+    def upload_to_gcs(cls, source_uri: str, target_uri: str):
+        cls.execute_with_ctx(
+            ["gsutil", "cp", f"{target_uri}", f"{source_uri}"], key=GCP_GCS_KEY
+        )
+
+    @classmethod
+    def upload_content_to_gcs(cls, lines: List[str], bucket_uri: str, filename: str):
```

Comment:
```suggestion
    def upload_content_to_gcs(cls, content: str, bucket_uri: str, filename: str):
```
WDYT?  I think it will be easier to use. Performance is not critical in this case. A simple types is also better.